### PR TITLE
Update the ADR naming convention and statuses 

### PR DESCRIPTION
**Pull request by andycochran**
## Summary
This PR is for a proposal discussed in Slack to help better organize ADRs and make creation simpler. We've already been forgetting to update statuses and the number in the filename before merging ADR PRs and having to create follow-up PRs. This proposes a new naming convention for ADRs and a simplified status and organization that makes clear which ADRs have been deprecated (which is currently none). 

1. **When creating an ADR file, use the date in the filename (not an incremented number).** 
    - This avoids race conditions (two people merge ADRs at the same time with the same number). The file list still puts the ADRs in order, and it gives us additional at-a-glance date info. If two ADRs have similar dates, that's more helpful than just knowing the explicit order they were (possibly) merged. 
2. **When drafting an ADR, just use the status the it will have once merged**
    - There's no reason to use a "Draft" status and have to remember to update it before merging. Being in a PR/branch implies that the ADR is a draft. If the ADR was merged, the decision was accepted. 
3. **Simplify ADR status to either "Active" or [deprecation reason]** 
    - “Accepted” is less meaningful. Even once deprecated, a decision was still “accepted” at one point, so it feels like the wrong word. “Active” implies the _decision is still being used_, not that the _ADR was approved_. 
4. **When an ADR is deprecated, update its status and move it to the `deprecated/` directory**
    - This removes the `Deprecated | ... | Superseded` status options, and prefers a simple explanation of why the ADR is no longer active. 
    - Open Q: Would it be possible for an ADR to be superseded but not deprecated? 

## Changes proposed

- Update the decisions `README` to clarify how ADRs should be organized
- Update the ADR template so that `status` simply communicates whether the ADR is active or deprecated (and why)
- Rename the ADR files using `YYMMDD` instead of incrementing a number

### CODE REVIEW

**Review by widal001 — APPROVED** [review]
I like these changes!! And it's an elegant solution to the potential race conditions. 

### Request
Would we be open to adding add `-` in the date portion of filename to make it a bit more readable? (e.g. convert `20230629-ci-cd-task-runner.md` to `2023-06-29-ci-cd-task-runner.md`)

If other folks are strongly opposed to that convention, I don't feel passionate enough about it to block the merge 😅

**Comment by SammySteiner** [comment]
I have a clarification question. The date refers to the date drafted, merged, or last updated?

**Comment by andycochran** [comment]
@widal001 Yup, I almost formatted it that way. Will change. 

@SammySteiner, re… 

> I have a clarification question. The date refers to the date drafted, merged, or last updated?

…TBH, I don't think it matters. We have a date in the metadata at the top of the file to track "Last Modified" and I don't think that needs to match the file name, even at the moment it's merged. I think the filename date will just give us a general idea of when something happened. If we need more exact timing, we can look at "Last Modified" and the PR merge date (which is more definitive and less prone to human error). I say we just give it today's date when making the file, and not worry about it.

**Comment by andycochran** [comment]
@SammySteiner, I just renamed all the files per @widal001's suggestion and realized we now have ADRs we've copied from the platform. This bros: 

image

So I used yesterday (the date they were merged) for the filename. Again, I don't think it matters all that much. We'll have a general order. And I think it's still better than an incremented number. These ADRs from the platform preexist this repo, and didn't make much sense as `0015` though `0018`. 

I could've dated these with older dates, but not all of them have dates in the metadata. And I think it's more meaningful (for these in particular) to be closer to the date _Grants_ made the decision, not when the _platform_ made the decision. 

Lemme know what y'all think about this slightly cavalier method of dating.

File: documentation/decisions/README.md [hunk 1]
```diff
@@ -44,10 +44,13 @@ While architecture decisions come in all shapes and sizes, some common examples
 When an architectural decision needs to be recorded, please use the following steps:
 
 1. [Create a new ADR ticket in GitHub](https://github.com/HHS/grants-equity/issues/new/choose)
-2. Identify the approvers for this decision, i.e. individuals or groups who must sign off on the decision before the ADR can be accepted (ideally this happens before work starts on the ADR)
+2. Identify the approvers for this decision, i.e. individuals or groups who must sign off on the decision before the ADR can be merged (ideally this happens before work starts on the ADR)
 3. Draft an ADR in a branch named after the ticket, then open a Pull Request (PR)
+  - Use the `{YYYY-MM-DD}-{description}.md` naming convention and copy the [template](./template.md) into the [adr](./adr/) directory (name the file with the date it was created, which may differ from the date the decision was made, PR was merged, or file was last modified)
```
Comment (andycochran) [inline documentation/decisions/README.md:49]: > The date refers to the date drafted, merged, or last updated?

Thinking more on this, I've updated the guidance to use the date when you create the file. That's most awkward when we're making ADRs retroactively, but it's the most straightforward method.

**Review by widal001 — APPROVED** [review]
Re-approving!

**Pull request merged**